joe/joe.py

```diff
@@ -88,25 +87,26 @@ def _fetch_gitignore(raw_name, directory=''):
     '''
     output = '\n#####=== %s ===#####\n' % raw_name
     if directory:
-        filepath = os.path.join(DATA_DIR, '%s' % directory + '/' + raw_name + \
-                                                 '.gitignore')
+        filepath = os.path.join(DATA_DIR, '%s/%s.gitignore' %
+                                (directory, raw_name))
     else:
         filepath = os.path.join(DATA_DIR, raw_name + '.gitignore')
         output += '\n'
     try:
-        with open(filepath) as f:
-            output += f.read()
+        with open(filepath) as gitignore:
```

Comment:
Pylint is a common standard for a reason. Variables shorter than three characters are generally a bad idea, with the exception of tiny iterators (`for i in foo:` but even then only in certain obvious cases). It's generally better to keep to a styling standard for an open source project rather than every coder to themselves. **Descriptive variables are better than undescriptive variables** (*as long as they aren't as long as many java variables*)